engine.go

```diff
@@ -216,20 +232,35 @@ func (engine *Engine) QuoteTo(buf *builder.StringBuilder, value string) {
 		return
 	}
 
-	if string(value[0]) == engine.dialect.QuoteStr() || value[0] == '`' {
-		buf.WriteString(value)
+	quotePair := engine.dialect.Quote("")
+
+	if value[0] == '`' || len(quotePair) != 2 || value[0] == quotePair[0] { // no quote
+		_, _ = buf.WriteString(value)
 		return
-	}
+	} else {
+		if len(quotePair) != 2 {
+			_, _ = buf.WriteString(value)
+			return
+		}
 
-	value = strings.Replace(value, ".", engine.dialect.QuoteStr()+"."+engine.dialect.QuoteStr(), -1)
+		prefix, suffix := quotePair[0], quotePair[1]
 
-	buf.WriteString(engine.dialect.QuoteStr())
-	buf.WriteString(value)
-	buf.WriteString(engine.dialect.QuoteStr())
+		_ = buf.WriteByte(prefix)
```

Comment:
The code is duplicated with them on `Quote`, could you create a new function for these?

Reply:
Oh, I see. Please wait